core/dbt/contracts/graph/manifest.py

```diff
@@ -1228,13 +1229,27 @@ def compatible_previous_versions(self):
             ("manifest", 8),
         ]
 
+    @classmethod
+    def upgrade_schema_version(cls, data):
+        # This is specific to manifests, so call this from ArtifactMixin code
+        if get_manifest_schema_version(data) <= 8:
+            data = upgrade_manifest_json(data)
+        return cls.from_dict(data)  # type: ignore
```

Comment:
That was just left over from the code that was moved out of VersionedSchema which could return a number of different classes. I can check to see if it can be removed.